pkg/auth/identity.go

```diff
@@ -0,0 +1,70 @@
+package auth
+
+import (
+	"context"
+
+	authapi "k8s.io/api/authorization/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Authorized map[string]bool
+
+type Identity struct {
+	Token   string
+	RestCfg *rest.Config
```

Comment:
Perhaps make the `RestCfg` not be a pointer since we want it to be copy that we can modify in `buildClient()`.